### PR TITLE
swww: init at 0.7.2

### PR DESCRIPTION
**Pull request by mateodd25**
###### Description of changes

Added swww, a package to change the wallpaper on wayland. See https://github.com/Horus645/swww for more details. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rsHalford** [comment]
Before merging this, could it get bumped to v0.6.0, please? Hopefully this gets merged soon!

**Comment by IogaMaster** [comment]
Please merge this

**Comment by IogaMaster** [comment]
I have created a basic flake for swww:
https://github.com/IogaMaster/swww-nix

**Comment by GlancingMind** [comment]
Swww just got a new release. https://github.com/Horus645/swww/releases/tag/v0.7.0

**Comment by IogaMaster** [comment]
Alright I have bumped it to v0.7.0

**Comment by mateodd25** [comment]
Updated to v0.7.0. I also hope this gets merged soon.

**Review by ocfox — COMMENTED** [review]
Suggest adding blank lines after the 12 and 16 lines.

File: pkgs/tools/wayland/swww/default.nix
Comment (adamcstephens) [inline pkgs/tools/wayland/swww/default.nix:2]:
```suggestion
, libxkbcommon }:
```

File: pkgs/tools/wayland/swww/default.nix
Comment (adamcstephens) [inline pkgs/tools/wayland/swww/default.nix:11]:
```suggestion
    rev = "refs/tags/v${version}";
    hash = "sha256-ofJTL+9izMmwENaEVYQ7rjpoEFR1J2szkJxzYb09N7g=";
```

File: pkgs/tools/wayland/swww/default.nix
Comment (adamcstephens) [inline pkgs/tools/wayland/swww/default.nix:17]: You should add yourself as maintainer of the package since I expect that was your goal when you added yourself to the global maintainers list.

File: pkgs/tools/wayland/swww/default.nix
Comment (adamcstephens) [inline pkgs/tools/wayland/swww/default.nix:19]:
```suggestion
    description = "Efficient animated wallpaper daemon for wayland, controlled at runtime";
```

File: pkgs/tools/wayland/swww/default.nix
Comment (adamcstephens) [inline pkgs/tools/wayland/swww/default.nix:15]: Have you tried to resolve the test failures? It looks like maybe it's just a path issue.

```
stderr=```"Error: \"error spawning swww-daemon: No such file or directory (os error 2)\"\n"```
command=`"/build/source/target/x86_64-unknown-linux-gnu/release/swww" "init"`
code=1
```

**Review by adamcstephens — CHANGES_REQUESTED** [review]
Some more whitespace would be nice.

The version in the tag is also mismatched with the release version. 

```
./result/bin/swww --version
swww 0.6.0
```

https://github.com/Horus645/swww/commit/d2f3570aa41707924abd4c0084850351580a21db#comments

File: pkgs/tools/wayland/swww/default.nix
Comment (mateodd25) [reply pkgs/tools/wayland/swww/default.nix:15]: I have not. I remember asking about this on discord when I first tried to build swww. Somebody pointed out that the issue is that the integration tests use the local network and are doomed to fail; see [https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md?plain=1#L331](documentation here). That said, I am not well-versed in nix or rust. So if you have an idea to solve it, I am happy to try it.

File: pkgs/tools/wayland/swww/default.nix
Comment (mateodd25) [reply pkgs/tools/wayland/swww/default.nix:17]: Done.

**Comment by mateodd25** [comment]
> Some more whitespace would be nice.
> 
Done.
> The version in the tag is also mismatched with the release version.
> 
> ```
> ./result/bin/swww --version
> swww 0.6.0
> ```
> 
> [Horus645/swww@d2f3570#comments](https://github.com/Horus645/swww/commit/d2f3570aa41707924abd4c0084850351580a21db#comments)

This should be fixed with release v0.7.1 (which, according to @Horus645, should be fairly soon). Should I wait for it or change the version to v0.6.0 and use the commit hash for rev?

File: pkgs/tools/wayland/swww/default.nix
Comment (adamcstephens) [reply pkgs/tools/wayland/swww/default.nix:16]: It also looks like the tests check for a functioning wayland environment, so probably best to disable them as you did here.

```
thread \'main\' panicked at \'Failed to connect to wayland environment: NoCompositorListening\', daemon/src/wayland.rs:51:45
```

**Comment by Horus645** [comment]
> This should be fixed with release v0.7.1 (which, according to @Horus645, should be fairly soon). Should I wait for it or change the version to v0.6.0 and use the commit hash for rev?

I am planning the new release for tomorrow, so you might want to wait one more day

**Comment by adamcstephens** [comment]
Result of `nixpkgs-review pr 202280` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swww</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 202280` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swww</li>
  </ul>
</details>

https://github.com/Horus645/swww/issues/59

Looks like aarch64 may need rustc 1.67.0, which is merged (#212811) but not yet available in master.

**Comment by mateodd25** [comment]
Sounds good. I don't have a computer with aarch64, so I cannot try it. But I am happy to wait if it means swww will reach more users.

**Comment by IogaMaster** [comment]
I bumped my flake to v0.7.1.

**Comment by adamcstephens** [comment]
@mateodd25 This now builds as rustc 1.67.0 is in master. Do you want to bump this PR to 0.7.1 and we can try and move forward?

**Comment by GlancingMind** [comment]
Another minor release dropped: https://github.com/Horus645/swww/releases/tag/v0.7.2

**Comment by IogaMaster** [comment]
I bumped my flake to v0.7.2

**Comment by mateodd25** [comment]
@adamcstephens Sorry for the delay; I was busy this week. I updated to v0.7.2. Let's go ahead and move forward.

**Review by adamcstephens — APPROVED** [review]
Result of `nixpkgs-review pr 202280` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swww</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 202280` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swww</li>
  </ul>
</details>

**Review by ocfox — APPROVED** [review]

**Review by robertodr — APPROVED** [review]

**Comment by mateodd25** [comment]
Sorry, I am not sure what the protocol is here. But what is missing for this PR to get merged?

**Comment by arjan-s** [comment]
Result of `nixpkgs-review pr 202280` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swww</li>
  </ul>
</details>

**Review by arjan-s — APPROVED** [review]

**Pull request merged**
